Here is the test that appears to be failing
```
test_getitem_cache (tests.test_dotty_cache.TestDottyCache) ... FAIL

======================================================================
FAIL: test_getitem_cache (tests.test_dotty_cache.TestDottyCache)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/build/qmk_dotty_dict-1.3.0.post1/tests/test_dotty_cache.py", line 14, in test_getitem_cache
    self.assertEqual(dot.__getitem__.cache_info().hits, 9)
AssertionError: 10 != 9
```

--- a/pkgs/development/python-modules/qmk-dotty-dict/default.nix
+++ b/pkgs/development/python-modules/qmk-dotty-dict/default.nix
@@ -1,0 +1,22 @@
+{ buildPythonPackage, fetchPypi, lib, pytest, setuptools-scm }:
+
+buildPythonPackage rec {
+  pname = "qmk_dotty_dict";
+  version = "1.3.0.post1";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-O2EeOTZgv6poNcaOlHhLroD+B7hJCXi17KsDoNL8fqI=";
+  };
+
+  nativeBuildInputs = [ setuptools-scm ];
+
+  doCheck = false;
+
+  meta = with lib; {
+    description = "Dictionary wrapper for quick access to deeply nested keys";
+    homepage = "https://github.com/pawelzny/dotty_dict";
+    license = licenses.mit;
+    maintainers = with maintainers; [ babariviere ];
+  };
+}

--- a/pkgs/tools/misc/qmk/default.nix
+++ b/pkgs/tools/misc/qmk/default.nix
@@ -1,18 +1,18 @@
-{ lib
-, python3
-, fetchpatch
-}:
+{ lib, python3, fetchpatch, writeText }:
 
 let
   inherit (python3.pkgs) buildPythonApplication fetchPypi;
-in
-buildPythonApplication rec {
+  setuppy = writeText "setup.py" ''
+    from setuptools import setup
+    setup()
+  '';
+in buildPythonApplication rec {
   pname = "qmk";
-  version = "0.0.52";
+  version = "1.0.0";
 
   src = fetchPypi {
     inherit pname version;
-    sha256 = "sha256-mNF+bRhaL6JhNbROmjYDHkKKokRIALd5FZbRt9Kg5XQ=";
+    sha256 = "sha256-2mLuxzxFSMw3sLm+OTcgLcOjAdwvJmNhDsynUaYQ+co=";
   };
 
   nativeBuildInputs = with python3.pkgs; [
@@ -27,7 +27,7 @@ buildPythonApplication rec {
     appdirs
     argcomplete
     colorama
-    dotty-dict
+    qmk-dotty-dict
     hid
     hjson
     jsonschema
@@ -35,6 +35,10 @@ buildPythonApplication rec {
     pygments
     pyusb
   ];
+
+  postConfigure = ''
+    cp ${setuppy} setup.py
+  '';
 
   # no tests implemented
   doCheck = false;
@@ -57,6 +61,6 @@ buildPythonApplication rec {
       - ... and many more!
     '';
     license = licenses.mit;
-    maintainers = with maintainers; [ bhipple ];
+    maintainers = with maintainers; [ bhipple babariviere ];
   };
 }

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -7563,6 +7563,8 @@ in {
 
   queuelib = callPackage ../development/python-modules/queuelib { };
 
+  qmk-dotty-dict = callPackage ../development/python-modules/qmk-dotty-dict {};
+
   r2pipe = callPackage ../development/python-modules/r2pipe { };
 
   rabbitpy = callPackage ../development/python-modules/rabbitpy { };

